adaptive/src/main/java/com/google/accompanist/adaptive/FoldAwareColumn.kt

```diff
@@ -0,0 +1,321 @@
+ * limitations under the License.
+ */
+
+package com.google.accompanist.adaptive
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.toComposeRect
+import androidx.compose.ui.layout.IntrinsicMeasurable
+import androidx.compose.ui.layout.IntrinsicMeasureScope
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.MeasurePolicy
+import androidx.compose.ui.layout.MeasureResult
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.layout.ParentDataModifier
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.platform.InspectorInfo
+import androidx.compose.ui.platform.InspectorValueInfo
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.window.layout.DisplayFeature
+import androidx.window.layout.FoldingFeature
+import kotlin.math.roundToInt
+
+/**
+ * A simplified version of [Column] that places children in a fold-aware manner.
+ *
+ * @param displayFeatures the list of known display features to automatically avoid
+ * @param modifier an optional modifier for the layout
+ * @param foldPadding the optional padding to add around a fold
+ * @param horizontalAlignment the horizontal alignment of the layout's children.
+ *
+ * @see [Column]
+ */
+@Composable
+public fun FoldAwareColumn(
+    displayFeatures: List<DisplayFeature>,
+    modifier: Modifier = Modifier,
+    foldPadding: PaddingValues = PaddingValues(),
+    horizontalAlignment: Alignment.Horizontal = Alignment.Start,
+    content: @Composable FoldAwareColumnScope.() -> Unit,
+) {
+    // Extract folding feature if horizontal and separating
+    val fold = displayFeatures.find {
+        it is FoldingFeature && it.orientation == FoldingFeature.Orientation.HORIZONTAL && it.isSeparating
+    } as FoldingFeature?
+
+    // Calculate fold bounds in pixels (including any added fold padding)
+    val foldBoundsPx = with(LocalDensity.current) {
+        val topPaddingPx = foldPadding.calculateTopPadding().roundToPx()
+        val bottomPaddingPx = foldPadding.calculateBottomPadding().roundToPx()
+
+        fold?.bounds?.toComposeRect()?.let {
+            Rect(
+                top = it.top - topPaddingPx,
+                left = it.left,
+                right = it.right,
+                bottom = it.bottom + bottomPaddingPx
+            )
```

Comment:
Do we prefer using `Rect` /`roundToPx` or `RectF` /`toPx`? Since these bounds are only used to determine if children overlap the fold, we're not limited to int values